### PR TITLE
MOD: handle Sync failure for ContactCommunications with null data

### PR DESCRIPTION
**Pull request by kti-daniel**

### CODE REVIEW

File: djconnectwise/sync.py [hunk 1]
```diff
@@ -296,6 +296,13 @@ def update_or_create_instance(self, api_instance):
                   " Error: {}".format(self.model_class, e)
             logger.error(msg)
             raise InvalidObjectException(msg)
+        except ValueError as e:
+            # ContactCommunication syncs don't fail with a record
+            # that has this kind of invalid data
+            msg = "ValueError while attempting to sync {}." \
+                  " Error: {}".format(self.model_class, e)
+            logger.error(msg)
```
Comment (kti-sam) [inline djconnectwise/sync.py:304]: This is not what I meant. We want to change the `ContactCommunication` model so it doesn't accept null or blank values at all. You need to change the actual model record:

From:
```
class ContactCommunication(models.Model):
value = models.CharField(blank=True, null=True, max_length=250)
```

To:
```
class ContactCommunication(models.Model):
value = models.CharField(max_length=250)
```

Comment (kti-sam) [reply djconnectwise/sync.py:304]: That way, if a record is saved with invalid data it is automatically caught with the current process.

Comment (kti-daniel) [reply djconnectwise/sync.py:304]: Oh, I need coffee this time. Thanks Sam.

Comment (kti-daniel) [reply djconnectwise/sync.py]: You can review this again

File: djconnectwise/models.py [hunk 2]
```diff
@@ -406,7 +406,7 @@ def default_fax_number(self):
 class ContactCommunication(models.Model):
     contact = models.ForeignKey(
         'Contact', null=True, on_delete=models.CASCADE)
-    value = models.CharField(blank=True, null=True, max_length=250)
+    value = models.CharField(default='', max_length=250)
```
Comment (kti-sam) [inline djconnectwise/models.py:409]: There should be no blank default here, as we want to get rid of anything null or blank. We can add a simple runpython operation in the migration to delete any invalid records (if they exist, but I doubt they do). Let me know if you would like any help with that.

**Pull request merged**